collections: add validation for collection name

SamiHiltunen opened the pull request:
* FIX Adds validation for collection names. Validation will be
  performed each time the name is assigned to.

* Improves code quality compliance.

Signed-off-by: Sami Hiltunen <sami.mikael.hiltunen@cern.ch>

tiborsimko commented on invenio_collections/models.py:
```diff
@@ -287,6 +318,12 @@ def ancestors(self):
             output |= c.dad.ancestors
         return output
 
+    def is_root(self):
```
I guess this definition of the root collection could lead to troubles.  

- It is true that there should be only one collection named CFG_SITE_NAME and that its collection ID should be 1.  This defines our "home collection".  

- However, you can have several "floating home collections" that people don't want to attach in the regular collection tree descending from the home collection.  E.g. imagine a "Drafts" collection that is restricted and searchable only when people use in the UI `cc=Drafts`.  Otherwise this collection wouldn't pop up in the facets or anywhere else, not being a descendant of the home collection.  

Using this definition of `is_root()`, we'd get True for both the Home collection and the Drafts collection, leading to name validation troubles.  The fact that a collection does not have any ancestors is not enough to make it the root node...

SamiHiltunen commented:
Added some tests and changed the is_root to check for whether the collection id == 1 instead of checking the number of ancestors. Also, I made is_root into a read-only property

jirikuncar commented on invenio_collections/tests/test_models.py:
```diff
@@ -0,0 +1,68 @@
+from invenio.testsuite import InvenioTestCase
+
+Collection = lazy_import('invenio_collections.models:Collection')
+cfg = lazy_import('invenio.base.globals:cfg')
```
`s/invenio\.base/invenio_base/`

jirikuncar commented:
@SamiHiltunen it looks good now. Please fix the import in tests and rebase. Thanks

SamiHiltunen commented:
@jirikuncar The import has been fixed and it is rebased now.

jirikuncar commented:
@SamiHiltunen please check the commit message.

SamiHiltunen commented:
@jirikuncar fixed the commit message
